webrender/src/clip.rs

```diff
@@ -280,10 +258,14 @@ pub struct ClipSources {
     pub local_outer_rect: Option<LayoutRect>,
     pub only_rectangular_clips: bool,
     pub has_image_or_line_decoration_clip: bool,
+    pub spatial_node_index: SpatialNodeIndex,
 }
 
 impl ClipSources {
-    pub fn new(clips: Vec<ClipSource>) -> Self {
+    pub fn new(
+        clips: Vec<ClipSource>,
```

Comment:
we could use `Iterator<Item = ClipSource>` to avoid heap allocation on the client

Reply:
It's not clear to me how this avoids an allocation in this case - could you expand on that?

Reply:
This function doesn't need a vec, since it goes through the elements and constructs other stuff from them. But it's currently requiring the caller to have that vec. If this function gets `IntoIterator` instead, the caller may still provide a vec, or they can just construct a complex iterator (with `chain()`, `zip()`, etc) that is completely on the stack. IIRC, there is at least one place in code where this would be handy.

Reply:
Ah, you're right - I misread that it took ownership of it, missing the ```collect```. Although, having tried to make that change it seems to flow on to changing several other functions, so perhaps better to leave for now (since this PR doesn't actually change that code), unless I'm missing an obvious way to make this change locally?

Reply:
It would be correct to change the calling functions as well, where possible, but at the same time `Vec` implements `IntoIterator` so those changes are not required. I'll be happy to follow up with this refactor ;)